sklearn/decomposition/_truncated_svd.py

```diff
@@ -212,6 +213,7 @@ def transform(self, X):
             Reduced version of X. This will always be a dense array.
         """
         X = check_array(X, accept_sparse='csr')
+        check_is_fitted(self)
```

Comment:
Should we check this before the `check_array`?

Reply:
Maybe, but I don't think it matter that much. `check_array` is indeed a bit faster, so if it's not fitted it'll fail faster. But check_array can also fail for incorrect input, in which case there is no need to check_is_fitted, and it doesn't take ages either. 

So overall +1 but I'm not sure it's worth making another PR for it.